### PR TITLE
 sdk/metric: Log empty meter name

### PR DESCRIPTION
**Pull request by pellared**
Fixes https://github.com/open-telemetry/opentelemetry-go/issues/4151

### CODE REVIEW

**Comment by pellared** [comment]
I do not think it is worth to mention logging in the changelog.

**Review by dmathieu — APPROVED** [review]

File: sdk/metric/provider.go [hunk 1]
```diff
@@ -68,6 +69,10 @@ func NewMeterProvider(options ...Option) *MeterProvider {
 //
 // This method is safe to call concurrently.
 func (mp *MeterProvider) Meter(name string, options ...metric.MeterOption) metric.Meter {
+	if name == "" {
+		global.Warn("Meter name should not be empty. Empty meter name is specified as invalid.")
```
Comment (MrAlias) [inline sdk/metric/provider.go:73]: This seems like a pretty verbose way to say the same thing as the following.

```suggestion
		global.Warn("Invalid Meter name: \"\"")
```

File: sdk/metric/provider_test.go [hunk 2]
```diff
@@ -74,3 +79,17 @@ func TestMeterProviderReturnsSameMeter(t *testing.T) {
...
 }
+
+func TestEmptyMeterName(t *testing.T) {
+	var buf strings.Builder
+	warnLevel := 1
+	l := funcr.New(func(prefix, args string) {
+		_, _ = buf.WriteString(fmt.Sprint(prefix, args))
+	}, funcr.Options{Verbosity: warnLevel})
+	otel.SetLogger(l)
+	mp := MeterProvider{}
```
Comment (MrAlias) [inline sdk/metric/provider_test.go:90]:
```suggestion
	mp := NewMeterProvider()
```

**Review by MrAlias — APPROVED** [review]

**Comment by pellared** [comment]
@MrAlias Can you just look once more? 😉

File: sdk/metric/provider.go [hunk 3]
```diff
@@ -68,6 +69,10 @@ func NewMeterProvider(options ...Option) *MeterProvider {
 //
 // This method is safe to call concurrently.
 func (mp *MeterProvider) Meter(name string, options ...metric.MeterOption) metric.Meter {
+	if name == "" {
+		global.Warn("Invalid Meter name.", "meterName", name)
```
Comment (MrAlias) [inline sdk/metric/provider.go:73]: I think from context the reader will know it is the meter name. Might be better to just leave it as "name"

```suggestion
		global.Warn("Invalid Meter name.", "name", name)
```

Comment (pellared) [reply sdk/metric/provider.go]: https://github.com/open-telemetry/opentelemetry-go/pull/4153/commits/a401554e6678098565974abd7462bf9e50c93490

**Review by MrAlias — APPROVED** [review]

**Pull request merged**